src/not-so-smart/protocol.ml

```diff
@@ -296,10 +296,6 @@ module Decoder = struct
         Fmt.pf ppf "Invalid result command (%S)" raw
     | `Unexpected_flush -> Fmt.string ppf "Unexpected flush"
 
-  let rec prompt_pkt ?strict k decoder =
-    if at_least_one_pkt decoder then k decoder
-    else prompt ?strict (prompt_pkt ?strict k) decoder
-
```

Comment:
Not sure about such deletion, I know that usually, I re-bind an OCaml value with the same name several such as:
```ocaml
let prompt_pkt = ...
let prompt_pkt = ... prompt_pkt ...
let prompt_pkt = ... prompt_pkt ...
...
```

So, in my mind, such deletion will disturb behaviors of functions defined then. Why you deleted this code?

Reply:
I moved `prompt_pkt` from `protocol.ml` to `decoder.ml` because 

1. `protocol.ml` contains `Decoder` module, which implements decoding for values of the pack protocol. But `prompt_pkt` is a function that is _not_ specific to protocol and could be defined on a lower and more general layer such as `decoder.ml`.
2. I also needed it in `proto_vals_v2.ml`, so it made sense to move this general and low-level function to `decoder.ml` which specifically deals with pkt-line decoding
3. Moving `prompt_pkt` to `decoder.ml` should be safe because all references to `prompt_pkt` in `protocol.ml` still refer to that value but "imported" from `decoder.ml`.